<launch>
    <arg name="pkg_name" value="auto_localization"/>
    <arg name="node_name" default="absolute_from_relative_position"/>

    <arg name="veh" doc="Name of vehicle. ex: megaman"/>

    <arg name="map" default="testcircle" doc="Specify the map you wants"/>
    <arg name="output" default="test" doc="Specify the output name of the statistics file"/>

    <arg name="tile" default="true" doc="Use tile related way to do global localization"/>

    <group ns="$(arg veh)">
        <group if="$(arg tile)">
          <node name="$(arg node_name)" pkg="$(arg pkg_name)" type="absolute_from_relative_position_tile.py"
                output="screen">
                <param name="~map" value="$(arg map)" type="str"/>
                <param name="~output_file" value="$(arg output)" type="str"/>
          </node>
        </group>
        <group unless="$(arg tile)">
          <node name="$(arg node_name)" pkg="$(arg pkg_name)" type="$(arg node_name).py"
                output="screen">
                <param name="~map" value="$(arg map)" type="str"/>
                <param name="~output_file" value="$(arg output)" type="str"/>
          </node>
        </group>
    </group>

</launch>
